client/app/components/queries/embed-code-dialog.js

```diff
@@ -9,7 +9,7 @@ const EmbedCodeDialog = {
 
     this.embedUrl = `${clientConfig.basePath}embed/query/${this.query.id}/visualization/${
       this.visualization.id
-    }?api_key=${this.query.api_key}`;
+    }?api_key=${this.query.api_key}&${location.search.substring(1)}`;
```

Comment:
I think parameters have a method for serializing themselves into a query string. Feels like a better option than this.

Reply:
Are you referring to [this](https://github.com/getredash/redash/blob/master/client/app/components/parameters.js#L19)? If so, feels too coupled to the directive's scope, no?

Reply:
Nope, this:

https://github.com/getredash/redash/blob/ade3cc72a70d2c65b6f8e393304f4a394f61e926/client/app/services/query.js#L165

Reply:
@arikfr It will return an object, not a string. Of course, it's not an issue to merge that params as we do in `Parameters` component (see the link @rauchy posted above), but we don't have any library to serialize them into query string (`$location.search(params)` does that job, but it does not expose any API).

Reply:
Ping :)

Reply:
@arikfr @kravets-levko Pardon [my NIH](https://github.com/getredash/redash/pull/3495/commits/17636ec6965336ce5af6f6620ca4f02bbcbe341d). I couldn't find a way to use $location on _other_ URLs so extending it with `Parameter.toUrlParams()` was of no help. LMK what you think of this.